api/v1alpha2/cloudprovider_types.go

```diff
@@ -25,49 +25,49 @@ limitations under the License.
 // support reflecting a struct with a field of type "map[string]TYPE" to INI.
 package v1alpha2
 
-// Config is the vSphere cloud provider's configuration.
-type Config struct {
+// CloudProviderConfig is the vSphere cloud provider's configuration.
+type CloudProviderConfig struct {
 	// Global is the vSphere cloud provider's global configuration.
 	// +optional
-	Global GlobalConfig `gcfg:"Global,omitempty" json:"global,omitempty"`
+	Global CloudProviderGlobalConfig `gcfg:"Global,omitempty" json:"global,omitempty"`
 
 	// VCenter is a list of vCenter configurations.
 	// +optional
-	VCenter map[string]VCenterConfig `gcfg:"VirtualCenter,omitempty" json:"virtualCenter,omitempty"`
+	VCenter map[string]CloudProviderVCenterConfig `gcfg:"VirtualCenter,omitempty" json:"virtualCenter,omitempty"`
 
 	// Network is the vSphere cloud provider's network configuration.
 	// +optional
-	Network NetworkConfig `gcfg:"Network,omitempty" json:"network,omitempty"`
+	Network CloudProviderNetworkConfig `gcfg:"Network,omitempty" json:"network,omitempty"`
 
 	// Disk is the vSphere cloud provider's disk configuration.
 	// +optional
-	Disk DiskConfig `gcfg:"Disk,omitempty" json:"disk,omitempty"`
+	Disk CloudProviderDiskConfig `gcfg:"Disk,omitempty" json:"disk,omitempty"`
 
 	// Workspace is the vSphere cloud provider's workspace configuration.
 	// +optional
-	Workspace WorkspaceConfig `gcfg:"Workspace,omitempty" json:"workspace,omitempty"`
+	Workspace CloudProviderWorkspaceConfig `gcfg:"Workspace,omitempty" json:"workspace,omitempty"`
 
 	// Labels is the vSphere cloud provider's zone and region configuration.
 	// +optional
-	Labels LabelConfig `gcfg:"Labels,omitempty" json:"labels,omitempty"`
+	Labels CloudProviderLabelConfig `gcfg:"Labels,omitempty" json:"labels,omitempty"`
 
-	// ProviderConfig contains extra information used to configure the
+	// CloudProviderProviderConfig contains extra information used to configure the
 	// vSphere cloud provider.
-	ProviderConfig ProviderConfig `json:"providerConfig,omitempty"`
+	ProviderConfig CloudProviderProviderConfig `json:"providerConfig,omitempty"`
 }
 
-// ProviderConfig defines any extra information used to configure
+// CloudProviderProviderConfig defines any extra information used to configure
 // the vSphere external cloud provider
-type ProviderConfig struct {
-	Cloud   *CloudConfig   `json:"cloud,omitempty"`
-	Storage *StorageConfig `json:"storage,omitempty"`
+type CloudProviderProviderConfig struct {
```

Comment:
I'm not sure how I much I like `CloudProviderProviderConfig `... Any suggestions to a different name that makes sense are more than welcome.